components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -722,8 +722,12 @@ protected void processAuthenticationResponse(HttpServletRequest request,
              * login page does not match. If the username submitted by login page is null, then the
              * username from the context will be considered as the username submitted from the login page.
              */
-             if (request.getParameter(USER_NAME) != null && usernameFromContext != null &&
-                    !usernameFromContext.equals(request.getParameter(USER_NAME))) {
+            String requestUsername = request.getParameter(USER_NAME);
+            String requestUsernameWithTenant =
+                    UserCoreUtil.addTenantDomainToEntry(requestUsername, requestTenantDomain);
+            if (requestUsername != null && usernameFromContext != null &&
```

Comment:
Shall we add a null check for requestUsernameWithTenant here?

Reply:
The requestTenantDomain being null or empty is handled within UserCoreUtil.addTenantDomainToEntry[1], therefore imo it is not needed here

[1] https://github.com/biliroy/carbon4-kernel/blob/master/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UserCoreUtil.java#L477

Reply:
As discussed offline, this is not necessary as there is an implicit null check within the String.equals method